docs: improve README and hows-to (#271)
eduNEXT/eox-core: README.rst

Eox-core is an `openedx plugin`_ for the `edx-platform`_, and part of the Edunext Open Extensions (aka EOX), that adds multiple API
endpoints to extend its functionality and avoid changing the base code directly. These
API endpoints include bulk creation of pre-activated users (for example, skip sending an activation email), enrollments, and pre-enrollment operations.

Installation
============

#. Add this plugin in your Tutor ``config.yml`` with the ``OPENEDX_EXTRA_PIP_REQUIREMENTS`` setting.

   .. code-block:: yaml
      
      OPENEDX_EXTRA_PIP_REQUIREMENTS:
         - eox-core=={{version}} # basic installation
         - eox-core{{requirements}}=={{version}} # requeriments e.g. [sentry,tpa]. Useful for integration with third-party applications.
         
#. Save the configuration with ``tutor config save``.
#. Build an open edx image with ``tutor images build openedx``.
#. Launch your platform with ``tutor local launch``.

**Note:** To use all the features, you need to have `the tutor-forum plugin <https://github.com/overhangio/tutor-forum>`_ and `the eox-tenant plugin <https://github.com/eduNEXT/eox-tenant>`_.

Features
=========

- Support redirections with middleware.

> - Add pipelines to be used with ``openedx-filters``.

Reviewer: I don't think this is actually true. The pipelines here are for the social auth authentication pipeline, not for openedx filters.

Reviewer: Could we update this line?

| Quince           | >= 10.0      |
+------------------+--------------+

⚠️ The Maple version does not support Django 2.2 but it does support Django 3.2 as of eox-core 7.0.

The plugin is configured for the latest release (Quince). The following changes in the plugin settings should be applied in order to be used for previous releases.

> If the release you are looking for is not listed, the actual configuration is sufficient or it is incompatible with the current eox-core version.

Reviewer 2: Maybe using _current_ instead of _actual_ is better in this  context? Here is a rewording suggestion:
```suggestion
If the release you are looking for is not listed, then either the current configuration is sufficient, or the release is incompatible with the current eox-core version.
```

**NOTE**: The Maple version does not support Django 2.2 but it does support Django 3.2 as of eox-core 7.0.

The plugin is configured for the latest release (Quince). The following changes in the plugin settings should be applied in order to be used for previous releases.

> If the release you are looking for is not listed, the actual configuration is sufficient or it is incompatible with the current eox-core version.

Reviewer: I thought that if wasn't not listed, it's not supported. In what case does the other scenario happen?

Author: Same explanation https://github.com/eduNEXT/eox-tenant/pull/210#discussion_r1650321069

Reviewer: Mmm, I get it now that you explained it. Back when I left the comment, I was thinking about the ambiguity of "or the configuration is enough or is not supported." How do I know?

Still, this is a non-blocking comment.

Author: Understanding that I add this line below

>  If you are installing a previous version, please refer to the tag to verify the configuration that should be applied.

I can think of a better way to explain it if it is still vague. :)

Reviewer: That sounds good!

Reviewer: We could follow this suggestion as well: https://github.com/eduNEXT/eox-tenant/pull/210#discussion_r1654992895, it seems clearer.

               profiles_sample_rate: 0.5
            another_client_parameter: 'value'

Auditing Django views
=====================

Most views in eox-core use an auditing decorator, defined in our custom library, *eox-audit-model*,
that helps save relevant information about non-idempotent operations. By default, this functionality is turned on. To
check your auditing records go to Django sysadmin and find DJANGO EDUNEXT AUDIT MODEL.

You can check the `eox-audit-model`_ documentation for more information.

Development
===========

Dependency Management
---------------------

> EOX core now follows `OEP-18`_ so the correct way to update dependencies is to run ``make upgrade`` inside your virtual environment.

Reviewer: Is this section needed? I haven't seen it in other repos.

Author: I didn't want to remove it without being sure if should keep it or not.

Reviewer: I'd say we don't need for a few reasons:
1. It was added back in 2020 while updating eox-core: https://github.com/eduNEXT/eox-core/blame/1176b68864847c2d3eec412e69e4c718106c161f/README.rst#L101 
2. That's the current standard for upgrading packages in a plugin, I don't think we need to specify it here

Reviewer: ^ Non-blocking comment.

.. _OEP-18: https://github.com/openedx/open-edx-proposals/blob/master/oeps/best-practices/oep-0018-bp-python-dependencies.rst

Testing
-------
You can find more information in the `Help for devs doc <https://github.com/eduNEXT/eox-core/blob/master/docs/help_for_devs/0001-include-test-cases-files.rst>`_.

.. _openedx plugin: https://github.com/openedx/edx-platform/tree/master/openedx/core/djangoapps/plugins
.. _edx-platform: https://github.com/openedx/edx-platform/
.. _eox-tenant: https://github.com/eduNEXT/eox-tenant/
.. _eox-audit-model: https://github.com/eduNEXT/eox-audit-model/

How to Contribute
=================

Contributions are welcome! See our `CONTRIBUTING`_ file for more
information – it also contains guidelines for how to maintain high code
quality, which will make your contribution more likely to be accepted.

.. _CONTRIBUTING: https://github.com/eduNEXT/eox-core/blob/master/CONTRIBUTING.rst

License
=======

> This software is licensed under the terms of the AGPLv3. See the LICENSE file for details.

Reviewer 3: ```suggestion
This software is licensed under the terms of the AGPLv3. See the `LICENSE <LICENSE.txt>`_ file for details.
```
